exceptions from native search scripts are no longer propagated to client

In ES-2.0.0-beta2 (and also in ES-2.0.0-beta1), when an exception in thrown from a native 
search script, Java node clients no longer seem to receive any indication that something went wrong.

For example, we have a native sort script that takes a parameter and throws an exception in case it is missing:
```java
    private static class MyFactory implements NativeScriptFactory {
        @Override
        public ExecutableScript newScript(final Map<String, Object> params) {
            if (params.get("field")== null) {
                throw new ScriptException("missing parameter: field");
            }
            ...
        }

        @Override
        public boolean needsScores() {
            return false;
        }
    }
```

however, the following integration test, that used to work with ES-1.7.1, now fails:
```java
public abstract class MyScriptIT extends ESIntegTestCase {
    @Rule
    public ExpectedException _exception = ExpectedException.none();

    @Test
    public void testMissingFieldFails() throws Exception {
        createIndex(...);
        ensureGreen(...);

        _exception.expect(SearchPhaseExecutionException.class);
        _exception.expectMessage("missing parameter: field");

        final Map<String, Object> params = new HashMap<>();
        final Script script = new Script("my-script", ScriptType.INLINE, "native", params);
        final ScriptSortBuilder script = SortBuilders.scriptSort(script, "number");
        return _client.prepareSearch(...).addSort(script).execute().actionGet();
    }
}
```

in fact, no exception at all is thrown
```
java.lang.AssertionError: Expected test to throw (an instance of org.elasticsearch.action.search.SearchPhaseExecutionException and exception with message a string containing "missing parameter: field")
...
```

in addition, it appears that 
- scripted fetch fields, scripted metrics also don't propagate any thrown exceptions.
  e.g. AggregationBuilders.sum("sum").script(...)
  e.g. AggregationBuilders.scriptedMetric(...)

- update scripts on the other hand still do return exceptions. 
  e.g. return _client.prepareUpdate("test", "test", id).setScript(...);
  However, the returned message always is "failed to execute script", 
  while it used to contain the message that was generated by our native script.

--- a/docs/reference/migration/migrate_2_0/java.asciidoc
+++ b/docs/reference/migration/migrate_2_0/java.asciidoc
@@ -32,6 +32,13 @@ Settings settings = Settings.settingsBuilder()
 Client client = TransportClient.builder().settings(settings).build();
 --------------------------------------------------
 
+==== Exception are only thrown on total failure
+
+Previously, many APIs would throw an exception if any shard failed to execute
+the request. Now the exception is only thrown if all shards fail the request.
+The responses for these APIs will always have a `getShardFailures` method that
+you can and should check for failures.
+
 ==== Automatically thread client listeners
 
 Previously, the user had to set request listener threads to `true` when on the
@@ -109,7 +116,7 @@ new InetSocketTransportAddress(new InetSocketAddress("127.0.0.1", 0));
 Elasticsearch used to shade its dependencies and to relocate packages. We no longer use shading or relocation.
 You might need to change your imports to the original package names:
 
-* `com.google.common` was `org.elasticsearch.common` 
+* `com.google.common` was `org.elasticsearch.common`
 * `com.carrotsearch.hppc` was `org.elasticsearch.common.hppc`
 * `jsr166e` was `org.elasticsearch.common.util.concurrent.jsr166e`
 * `com.fasterxml.jackson` was `org.elasticsearch.common.jackson`
@@ -121,4 +128,3 @@ You might need to change your imports to the original package names:
 * `com.tdunning.math.stats` was `org.elasticsearch.common.stats`
 * `org.apache.commons.lang` was `org.elasticsearch.common.lang`
 * `org.apache.commons.cli` was `org.elasticsearch.common.cli.commons`
- 

